android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListItemProxy.java

```diff
@@ -301,6 +260,202 @@ public void call(Object data)
+		// Note: Given "proxy" shouldn't have any child proxies, but do the copy just in case.
+		copyChildPropertiesFromProxies();
+		proxy.copyChildPropertiesFromProxies();
+
+		// Remove all children from given proxy. (If recycling is working as expected, it should have none.)
+		proxy.removeAllChildren();
+
+		// Do not continue if templates don't match. Prevents given proxy from re-using this proxy's views.
+		// Note: Should never happen if ListViewAdapter.getItemViewType() returns different IDs for different templates.
+		String templateId = getTemplateId();
+		if ((templateId == null) || !templateId.equals(proxy.getTemplateId())) {
+			return;
+		}
+
+		// Do not continue if this proxy has no children.
+		// This will force given proxy to regenerate child proxies from template.
+		TiViewProxy[] childProxies = getChildren();
+		if ((childProxies == null) || (childProxies.length <= 0) || (childProxies[0].peekView() == null)) {
+			return;
+		}
+
+		// Move this proxy's children to given proxy and overwrite their properties.
+		// Note: This also moves their native views. Updating properties will also update the native views.
+		proxy.loadTemplate();
+		proxy.copyChildPropertiesTo(childProxies, proxy.template);
+		proxy.add(childProxies);
+
+		// Now that this proxy has no children, release this proxy's 1 native view container.
+		// We do this to reduce memory footprint of all offscreen list items.
+		releaseViews();
+
+		// Remove all child binding IDs from this proxy except for the ListItem itself.
+		final String bindId = this.template.getString(TiC.PROPERTY_BIND_ID);
+		this.binds.clear();
+		if (bindId != null) {
+			this.binds.put(bindId, this);
+		}
+	}
+
+	private void copyChildPropertiesFromProxies()
+	{
+		for (HashMap.Entry<String, TiViewProxy> bindsEntry : this.binds.entrySet()) {
+			// Fetch the next child view proxy. (Skip the root ListItemProxy object.)
+			TiViewProxy proxy = bindsEntry.getValue();
+			if (proxy == this) {
+				continue;
+			}
+
+			// Do no continue if child proxy has no properties.
+			KrollDict proxyProperties = proxy.getProperties();
+			if (proxyProperties == null) {
+				continue;
+			}
+
+			// Fetch properties dictionary to store child view's changes to.
+			HashMap cachedProperties = null;
+			Object object = this.childProperties.get(bindsEntry.getKey());
+			if (object instanceof HashMap) {
+				cachedProperties = (HashMap) object;
+			} else {
+				cachedProperties = new KrollDict();
+				this.childProperties.put(bindsEntry.getKey(), cachedProperties);
+			}
+
+			// Copy all view proxy properties to this list item's dictionary.
+			// Saves view's current state when scrolled offscreen so we can restore it when scrolled back in.
+			for (KrollDict.Entry<String, Object> proxyPropertyEntry : proxyProperties.entrySet()) {
+				cachedProperties.put(proxyPropertyEntry.getKey(), proxyPropertyEntry.getValue());
+			}
```

Comment:
Looks like this could be simplified a little
```Java
// Fetch properties dictionary to store child view's changes to.
HashMap cachedProperties = null;
Object object = this.childProperties.get(bindsEntry.getKey());
if (object instanceof HashMap) {
    cachedProperties = (HashMap) object;
    cachedProperties.putAll(proxyProperties);
} else {
    cachedProperties = new KrollDict(proxyProperties);
    this.childProperties.put(bindsEntry.getKey(), cachedProperties);
}
```

Reply:
Yeah, I agree with you. And doing a `putAll()` should be faster too.